### PR TITLE
Don't save unchanged script upon closing

### LINKED ISSUE #46528: Closing a script triggers unwanted save of that script
**Godot version:**
Godot 3.2.4 rc3

**OS/device including version:**
Windows 10 64 bits

**Issue description:**
When I close a script from the editor using `CTRL+W`, Godot saves that script without my consent, even if that script was not modified.

When the script is `tool`, it has the consequence of triggering reloading, which almost always breaks plugins. So I ended up in a loop of being unable to close open scripts while testing my plugin because it constantly broke it, and it took me a while to understand why that happened.

Closing a script should not be a modification operation. It should just close the script. Saving should happen only after the user confirmed to do so (which is the case when the script is modified with a popup).

**Steps to reproduce:**
1) Open a project which had open scripts (or open one, but I havent tested this way)
2) Close it with Ctrl+W
3) Observe the file modification date changes

It happens probably because of this code in `script_editor_plugin.cpp`:
```cpp
			case FILE_CLOSE: {
				if (current->is_unsaved()) {
					_ask_close_current_unsaved_tab(current);
				} else {
					_close_current_tab();
				}
```
`_close_current_tab()` is implemented like this:
```cpp
void ScriptEditor::_close_current_tab() {

	_close_tab(tab_container->get_current_tab());
}
```
And `_close_tab` has the following default arguments:
```cpp
	void _close_tab(int p_idx, bool p_save = true, bool p_history_back = true);
```
It saves by default. It should not.

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #46528

### CODE REVIEW

File: editor/plugins/script_editor_plugin.h [hunk 1]
```diff
@@ -325,7 +325,8 @@ class ScriptEditor : public PanelContainer {
 
 	void _close_tab(int p_idx, bool p_save = true, bool p_history_back = true);
 
-	void _close_current_tab();
+	void _close_save_current_tab() { _close_current_tab(); } // callable_mp can't use default arguments :/
```
Comment (akien-mga) [inline editor/plugins/script_editor_plugin.h:328]: `callable_mp` can't, but `connect` takes parameters. So you could have used `erase_tab_confirm->connect("confirmed", callable_mp(this, &ScriptEditor::_close_current_tab), varray(true));`.

Comment (KoBeWi) [reply editor/plugins/script_editor_plugin.h]: Right...

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.4.

**Pull request merged**